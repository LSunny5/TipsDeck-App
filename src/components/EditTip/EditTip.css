  .editContent {
    padding-bottom: 3rem;
  }
  
  .editForm {
    margin: 3px;
    text-align: left;
    padding: 0 20px;
    font-size: 1em;
    display: inline-block;
    height: 100%;
  }

  .inputEdit {
    margin: 10px;
    border: 1px solid black;
    font-size: 1em;
    width: 55%;
    text-align: left;
    padding-left: 5px;
  }
  
  select {
    width: 50%;
    margin: 0 0 0 15px;
  }
  
  Textarea{
    border: 1px solid black;
    height: 12vh;
    width: 100%;
    font-size: 1em;
    text-align: left;
  }
  
  .rateCSS {
    width: 25%;
    border: 1px solid black;
    padding: 5px; 
  }

  .buttonBox {
    text-align: left;
    margin-top: 10px;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
  }
  
  .editButton {
    border: 1px solid black;
    box-shadow: 0;
    padding: 0.5rem;
    width: 40%;
    font: 15px;
    text-decoration: none;
    color: black;
    margin-bottom: 2rem;
  }

  /*for ipads and larger screen sizes*/
   @media only screen and (min-width: 600px) {
    .editForm {
      width: 80%;
      margin: 1% auto;
      padding: 0 5%;
      font-size: 1.3em;
    }

    .inputEdit {
      padding: 1%;
      width: 65%;
    }

    .inputLabel {
      font-size: 1.2em;
      font-weight: bold;
      color: darkblue;
    }

    Textarea{
      height: 10vh;
      font-size: 1.2em;
    }

    #sourceName {
      width: 62%;
    }

    .rateCSS {
      font-size: 1em;
      width: 15%;
      margin: 10px;
    }

    .editButton {
      padding: 5%;
      height: 7vh;
      margin: 3%;
      width: 30%;
      font-weight: bolder;
      margin-bottom: 5rem;
    }
  }
  
  /*for desktops*/
  @media only screen and (min-width: 1024px) {
    .editForm {
      font-size: 1em;
    }

    Textarea{
      height: 8vh;
      font-size: 1.2em;
    }

    .editButton {
      padding: 2%;
      height: 7vh;
      margin: 2% auto;
      width: 20%;
      margin-bottom: 2rem;
    }
  }