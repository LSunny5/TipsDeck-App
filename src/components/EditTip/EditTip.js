import React from 'react';
import './EditTip.css';
import TipsDeckContext from '../../TipsDeckContext';
import { findTip, getCategoryName } from '../../TipsDeckHelpers';
import { NavLink } from 'react-router-dom';

//Edit the Tip Information
class EditTip extends React.Component {
    static contextType = TipsDeckContext;
    constructor(props) {
        super(props);
        this.state = {
            TipName: {
                value: '',
                touched: false
            },
            Category: {
                value: '',
                touched: false
            },
            descriptions: {
                value: '',
                touched: false
            },
        }
    }

    updateName(name) {
        this.setState({ TipName: { value: name, touched: true } });
    }

    updateCategory(category) {
        this.setState({ Category: { value: category, touched: true } });
    }

    updateDesc(desc) {
        this.setState({ description: { value: desc, touched: true } });
    }

   




    handleChange = event => {
		const { name, value } = event.target;
		this.setState({ [name]: value });
	};


    handleClickEdit = event => {
        event.preventDefault();
        const {id, name} = this.props.match.params;

        const changedNote = {
            category: event.target['Category'].value,
            name: event.target['TipName'].value,
            description: event.target['description'].value,
           /*  directions: this.state.directions,
            sourceURL: this.state.sourceURL, */
        }

        this.context.editTip(changedNote);
        this.props.history.push(`/Category/${name}/${id}`);


        //this.props.history.push('/');
    }

    handleClickCancel = () => {
        //this.props.history.goBack();
        const {id, name} = this.props.match.params;
        this.props.history.push(`/Category/${name}/${id}`);
        alert('Tip was not edited.');
    }

    render() {
        const { tips = [], categories = [] } = this.context;
        const { id } = this.props.match.params;

        const tipEdit = findTip(tips, id) || {};
        const catForTip = getCategoryName(categories, tipEdit.category) || {};

        return (
            <section className="editContent">
                <h1>Edit Tip</h1>
                <h2>"{tipEdit.name}"</h2>
                <form className="editForm" onSubmit={this.handleClickEdit}>
                    <label htmlFor="TipName" className="inputLabel">Name of Tip: </label>
                    <input
                        type="text"
                        id="TipName"
                        name="TipName"
                        className="inputEdit"
                        required
                        defaultValue={tipEdit.name}
                        onChange={e => this.updateName(e.target.value)}
                    />
                    <br />
                    <label htmlFor="Category" className="inputLabel">Category: </label>
                    <select id="Category" className="inputEdit"
                        defaultValue={catForTip.id}
                        required
                        onChange={e => this.updateCategory(e.target.value)}
                    >
                        {categories.slice(0, categories.length - 1).map(category =>
                            <option
                                key={category.id}
                                value={category.name}
                            >
                                {category.name}
                            </option>
                        )}
                    </select>
                    <br />
                    <label htmlFor="description" className="inputLabel">Description: </label>
                    <br />
                    <textarea id="description" name="description" defaultValue={tipEdit.description} onChange={e => this.updateDesc(e.target.value)}></textarea>
                    <br />
                    <label htmlFor="directions" className="inputLabel">Directions: </label>
                    <br />
                    <textarea id="directions" name="directions" defaultValue={tipEdit.directions} onChange={this.handleChange}></textarea>
                    <br />
                    <label htmlFor="source" className="inputLabel">Source URL: </label>
                    <textarea
                        type="text"
                        id="source"
                        name="source"
                        className="sourceText"
                        defaultValue={tipEdit.sourceURL}
                        onChange={this.handleChange}
                    />
                    <br />
                    <div className="buttonBox">
                        <button 
                            type="submit" 
                            className="editButton"
                            /* onClick={() => {
                                if (window.confirm('Are you sure you wish to update the tip?') ? this.handleClickEdit() : this.handleClickCancel())
                                    this.handleClickCancel()
                            }} */
                        >
                            Save Changes
                        </button>
                        <NavLink
                            className="editButton" 
                            to={`/Category/${catForTip.name}/${tipEdit.id}`}
                        >
                            Cancel
                        </NavLink>
                    </div>


                </form>
            </section>
        );
    }
}

export default EditTip;
