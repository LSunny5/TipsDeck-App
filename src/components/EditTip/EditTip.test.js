import React from 'react';
import ReactDOM from 'react-dom';
import EditTip from './EditTip';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`EditTip component`, () => {
    const props = {
        match: {
          params: {
            category: 'Laundry'
          }
        }, 
        category_id: 1,
            tipname: 'test tip jest 1',
            tipdescription: 'description for test tip jest 1',
            directions: 'directions for test tip jest 1',
            sourcetitle: 'title for test tip jest 1',
            sourceurl: 'url for test tip jest 1',
            rating: 0,
            numraters: 0,
            history: {
                push: () => { }
            }
      }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <EditTip {...props} />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the EditTip form', () => {
        const wrapper = shallow(<EditTip {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});