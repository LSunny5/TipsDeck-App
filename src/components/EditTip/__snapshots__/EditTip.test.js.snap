// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTip component renders the EditTip form 1`] = `
<section
  className="editContent"
>
  <form
    className="formBox"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Edit Tip
      </legend>
      <h2>
        "
        "
      </h2>
      <label
        className="inputLabel"
        htmlFor="TipName"
      >
        Name of Tip 
        <span
          className="required"
        >
          *
        </span>
        :  
      </label>
      <input
        className="inputEdit"
        id="TipName"
        name="TipName"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <div
        className="errorBox"
        id="contentErrorBox"
      />
      <label
        className="inputLabel"
        htmlFor="Category"
      >
        Category: 
      </label>
      <select
        className="inputEdit"
        id="Category"
        onChange={[Function]}
        required={true}
      />
      <br />
      <label
        className="inputLabel"
        htmlFor="description"
      >
        Description 
        <span
          className="required"
        >
          *
        </span>
        : 
      </label>
      <br />
      <textarea
        id="description"
        name="description"
        onChange={[Function]}
      />
      <div
        className="errorBox"
        id="contentErrorBox"
      />
      <label
        className="inputLabel"
        htmlFor="directions"
      >
        Directions: 
      </label>
      <textarea
        id="directions"
        name="directions"
        onChange={[Function]}
      />
      <br />
      <label
        className="inputLabel"
        htmlFor="sourceName"
      >
        Source Name 
        <span
          className="required"
        >
          *
        </span>
        : 
      </label>
      <input
        className="inputEdit"
        id="sourceName"
        name="sourceName"
        onChange={[Function]}
        required={true}
        type="text"
      />
      <div
        className="errorBox"
        id="contentErrorBox"
      />
      <label
        className="inputLabel"
        htmlFor="sourceurl"
      >
        Source URL: 
      </label>
      <textarea
        className="sourceText"
        id="sourceurl"
        name="sourceurl"
        onChange={[Function]}
        type="text"
      />
      <div
        className="rateBox"
      >
        <div>
          <label
            className="inputLabel"
            htmlFor="rating"
          >
            Rating (0-5) 
            <span
              className="required"
            >
              *
            </span>
            :  
          </label>
          <input
            className="rateCSS"
            id="rating"
            name="rating"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div>
          <label
            className="inputLabel"
            htmlFor="raters"
          >
            # of Raters: 
          </label>
          <input
            className="rateCSS"
            id="raters"
            name="raters"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
      </div>
      <div
        className="errorBox"
        id="contentErrorBox"
      />
      <div
        className="formButtonBox"
      >
        <button
          className="formSubmit formButton"
          type="submit"
        >
          Save Changes
        </button>
        <NavLink
          className="formCancel formButton"
          to="/Category/undefined/undefined"
        >
          Cancel
        </NavLink>
      </div>
    </fieldset>
  </form>
</section>
`;
